### ISSUE TITLE
Focal Loss for Multi-Label Text Classification

### ISSUE THREAD

**Issue by amitbcp** [flairNLP/flair#806]
When we use Flair for Multi-Label Text Classification, we use the BCELoss. This works great but suffers badly in cases of class-imbalances in the data.
An alternative for the same could be FocalLoss defined an release by Facebook.

Focal loss is a Cross-Entropy Loss that weighs the contribution of each sample to the loss based in the classification error. The idea is that, if a sample is already classified correctly by the CNN, its contribution to the loss decreases. With this strategy, they claim to solve the problem of class imbalance by making the loss implicitly focus in those problematic classes.
Moreover, they also weight the contribution of each class to the lose in a more explicit class balancing.

Paper Reference  : https://arxiv.org/abs/1708.02002
Sample Code for Focal Loss in PyTorch - https://www.kaggle.com/c/tgs-salt-identification-challenge/discussion/65938

It can be included as an experiment, as many issues locked on the project face issue in Text Classification

**Comment by alanakbik** [comment]
Interesting - yes we should try it out!

**Comment by amitbcp** [comment]
Hi @alanakbik ,

I remember seeing a PR where Focal Loss implementation was implemented but the removed before the merge . I thought it would make it to 0.4.3 but don't see it.
What are the plans for it?
